akka-cluster/src/main/scala/akka/cluster/MembershipState.scala

```diff
@@ -119,4 +124,137 @@ import akka.annotation.InternalApi
       .map(_.uniqueAddress)
   }
 
+  def validNodeForGossip(node: UniqueAddress): Boolean =
+    node != selfUniqueAddress && isReachableExcludingDownedObservers(node)
```

Comment:
Ah, so actual pair wise reachability observations then, hmm. Well as failure detection should happen from the same N node I guess if link is down there will be exactly those pairs?

That invalidates my previous thought about limiting to the N first gossipable nodes though.